iphone/Classes/GeolocationModule.m

```diff
@@ -794,18 +805,26 @@ - (void)requestLocationPermissions:(CLAuthorizationStatus)authorizationType with
     [callback.context.virtualMachine addManagedReference:authorizationCallback withOwner:self];
   }
 
+  // Promise to return
+  KrollPromise *promise = [[KrollPromise alloc] initInContext:[self currentContext]];
```

Comment:
Memory leak. Same as above.

Reply:
It's explicitly retained and assigned to `authorizationPromise` a few lines below this.

Reply:
Yes, it should be autoreleased.